js/controller.js

```diff
@@ -335,18 +338,27 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
      * @description
      * Should be used with deviceorientation event listener.
      * Uses for video 360 on mobile devices for setting necessary coordinates (relevant with start device orientation)
+     * Before playing this.vrMobileOrientationChecked is equal false,
+     * if need to check value for device orientation set this.checkDeviceOrientation = true
      * @param {object} e The event object
      */
     handleVrMobileOrientation: function(e) {
-      if (!this.vrMobileOrientationChecked) {
+      if (!this.vrMobileOrientationChecked || this.checkDeviceOrientation) {
         var beta = e.beta;
+        var gamma = e.gamma;
         var yaw = this.state.vrViewingDirection["yaw"];
         var pitch = this.state.vrViewingDirection["pitch"];
-        if (beta !== undefined && beta !== null && Utils.ensureNumber(beta, 0)) {
-          pitch += -90 + Math.round(beta);
+        var dir = beta;
+        var orientationType = Utils.getOrientationType();
+        if (orientationType && (orientationType === "landscape-secondary" || orientationType === "landscape-primary")) {
```

Comment:
Should we create an isLandscape helper function, checking orientationType === "landscape-secondary" || orientationType === "landscape-primary"?

Reply:
This is not necessary because it is not used in any other places.